sotodlib/io/bookbinder.py

```diff
@@ -153,19 +210,22 @@ def add(self, f):
         self.times.extend(f['data'].times)
 
         if self.signal is None:
-            self.signal = self.create_new_supertimestream(f['data'], use_rids=True)
+            self.signal = so3g.G3SuperTimestream(f['data'].names, core.G3VectorTime([]),
+                                    np.empty((len(f['data'].names), 0), dtype=np.int32))
         self.signal.times.extend(f['data'].times)
         self.signal.data = np.hstack((self.signal.data, f['data'].data)).astype(np.int32)
 
         if 'tes_biases' in f.keys():
-            if self.bias is None:
-                self.bias = self.create_new_supertimestream(f['tes_biases'])
-            self.bias.times.extend(f['tes_biases'].times)
-            self.bias.data = np.hstack((self.bias.data, f['tes_biases'].data)).astype(np.int32)
+            if self.biases is None:
+                self.biases = so3g.G3SuperTimestream(f['tes_biases'].names, core.G3VectorTime([]),
+                                    np.empty((len(f['tes_biases'].names), 0), dtype=np.int32))
+            self.biases.times.extend(f['tes_biases'].times)
+            self.biases.data = np.hstack((self.biases.data, f['tes_biases'].data)).astype(np.int32)
```

Comment:
I guess the performance fix will go to another PR, but I'm a little confused about the mixed use of int32 and int64 across different fields. Could you clarify that for me?

Reply:
Those are the types in the original SMuRF files (int32 for `'data'` and `'tes_biases'` and int64 for `'primary'`). I did ask @mhasself at one point about the discrepancy with the TOD format doc, and then we forgot to update it. Nonetheless, I plan to add type-checking assertions in a subsequent PR where I optimize the array type-casting.

Reply:
@jlashner  will know what's actually coming out of the smurf system.

Reply:
This is correct, the TES signal and bias data are both 32 bit ints, while the "primary" data which contains timing counters and timestamps are 64 bit.